packages/web-components/src/components/va-banner/va-banner.tsx

```diff
@@ -0,0 +1,163 @@
+
+  /**
+   * Keep track of locally dismissed Banners
+   * */
+  @State() dismissedBanners: Array<String> = [];
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link or the dismiss icon is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private prepareBannerID = () => `${this.headline}:${this.el.innerHTML}`;
+
+  private dismiss = () => {
+    // Derive the current banner ID.
+    const currentBannerID = this.prepareBannerID();
+
+    // Escape early if the banner is already dismissed.
+    if (this.dismissedBanners.includes(currentBannerID)) {
+      return;
+    }
+
+    // Add the banner ID to the list of dismissed banners. If showClose enabled
+    if (this.showClose) {
+      // Derive the updated dismissed banners.
+      const updatedDismissedBanners = [
+        ...this.dismissedBanners,
+        currentBannerID,
+      ];
+
+      //   Set storage type
+      const storageType = this.windowSession
+        ? window.sessionStorage
+        : window.localStorage;
+
+      // Set banner dismissed in storage
+      storageType.setItem(
+        DISMISSED_BANNERS_KEY,
+        JSON.stringify(updatedDismissedBanners),
+      );
+
+      // Update dismissedBanners in state.
+      this.dismissedBanners = updatedDismissedBanners;
+
+      // Track the dismiss event in Google Analytics
+      if (!this.disableAnalytics) {
+        const detail = {
+          componentName: 'va-banner',
+          action: 'close',
+          details: {
+            headline: this.headline,
+          },
+        };
+        this.componentLibraryAnalytics.emit(detail);
+      }
+    }
+  };
+
+  // Called once just after the component is first connected to the DOM
+  // Fixes rerender issues that could be potentially created by calling dismissedBanners multiple times
+  componentWillLoad() {
+    // Only fire if showClose enabled
+    if (this.showClose) {
+      // Set the storage type
+      const storageType = this.windowSession
+        ? window.sessionStorage
+        : window.localStorage;
+
+      // Derive dismissed banners from storage.
+      const dismissedBannersString = storageType.getItem(DISMISSED_BANNERS_KEY);
+      this.dismissedBanners = dismissedBannersString
+        ? JSON.parse(dismissedBannersString)
+        : [];
+    }
+  }
+
+  render() {
+    // Derive if the banner is dismissed.
+    const isBannerDismissed =
+      this.showClose && this.dismissedBanners?.includes(this.prepareBannerID());
+
+    // Escape early if the banner isn't visible or is dismissed.
+    if (!this.visible || isBannerDismissed) {
+      return null;
+    }
+
+    // Derive onCloseAlert depending if the close icon is shown.
+    const onCloseAlert = this.showClose ? this.dismiss : undefined;
+
+    return (
+      <Host
+        data-e2e-id="emergency-banner"
```

Comment:
This might not be necessary anymore. It may have been in `<Banner>` originally for some nightwatch tests, but I'm not seeing `emergency-banner` used anywhere in `vets-website` cypress tests.

Reply:
Excellent going to remove